### ISSUE TITLE
cri-o 1.17.3 package doesnt exist anymore and v1.17.4 breaks minikube 

### ISSUE THREAD

**Issue by medyagh** [cri-o/cri-o#4027]
hello,

I would like to know how I could install cri-o v1.17.3 , since the package has vanished from [the repo](https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/xUbuntu_19.10/).

in minikube project, we faced a blocker to build the minikube base image since we can no longer install cri-o v1.17.3.
and unfortunately cri-o v1.17.4 doesn't not pass our tests.  https://github.com/kubernetes/minikube/issues/8861

we have invested a lot of testing infrastructure and engineering time to provide cri-o as a runtime in minikube, I would greatly apperciate it if you could assist us to have an easier way to install cri-o without a lot of efforts and suprises.

here is how we are installing it, if that helps to guide us better:

```

# install cri-o based on https://github.com/cri-o/cri-o/commit/96b0c34b31a9fc181e46d7d8e34fb8ee6c4dc4e1#diff-04c6e90faac2675aa89e2176d2eec7d8R128
RUN sh -c "echo 'deb http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/xUbuntu_19.10/ /' > /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list" && \    
    curl -LO https://download.opensuse.org/repositories/devel:kubic:libcontainers:stable/xUbuntu_19.10/Release.key && \
    apt-key add - < Release.key && apt-get update && \
    apt-get install -y --no-install-recommends cri-o-1.17

```

**Comment by haircommander** [comment]
hey @medyagh 

Sorry for the frustration. I am not sure where 1.17.3 went, but I have attempted to package it (this is my first rodeo, so bear with me if there are any difficulties).

If you want to pin to a specific version, you can specify the subproject as so:
https://build.opensuse.org/repositories/devel:kubic:libcontainers:stable:cri-o:1.17:1.17.3

as oppposed to just `1.17` or pulling cri-o from the stable repo directly. 

Right now, we're packaing for upstream as best we can, but we're a fairly small team who have a lot of requests flying at us. OBS is the best bet, as this is where we're putting our energy, unless you install from static binaries available at the releases page.

**Comment by medyagh** [comment]
@haircommander thanks for the response,
I tried this in our Dockerfile and it didn't build

```
RUN sh -c "echo 'deb https://build.opensuse.org/repositories/devel:kubic:libcontainers:stable:cri-o:1.17:1.17.3 /' > /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list" && \    
    curl -LO https://download.opensuse.org/repositories/devel:kubic:libcontainers:stable/xUbuntu_19.10/Release.key && \
    apt-key add - < Release.key && apt-get update && \
    apt-get install -y --no-install-recommends cri-o-1.17
```


here is the full log from the docker build
```
 ---> Running in 5fab3dc9c587
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   402  100   402    0     0    728      0 --:--:-- --:--:-- --:--:--   726
100   403  100   403    0     0    560      0 --:--:-- --:--:-- --:--:--   560
100   404  100   404    0     0    453      0 --:--:-- --:--:-- --:--:--   453
100  1094  100  1094    0     0   1031      0  0:00:01  0:00:01 --:--:--  1031
Warning: apt-key output should not be parsed (stdout is not a terminal)
OK
Hit:1 http://security.ubuntu.com/ubuntu eoan-security InRelease
Hit:2 http://archive.ubuntu.com/ubuntu eoan InRelease
Hit:3 http://archive.ubuntu.com/ubuntu eoan-updates InRelease
Hit:4 http://archive.ubuntu.com/ubuntu eoan-backports InRelease
Ign:5 https://build.opensuse.org/repositories/devel:kubic:libcontainers:stable:cri-o:1.17:1.17.3  InRelease
Err:6 https://build.opensuse.org/repositories/devel:kubic:libcontainers:stable:cri-o:1.17:1.17.3  Release
  404  Not Found [IP: 195.135.221.162 443]
Reading package lists...
E: The repository 'https://build.opensuse.org/repositories/devel:kubic:libcontainers:stable:cri-o:1.17:1.17.3  Release' does not have a Release file.
The command '/bin/sh -c sh -c "echo 'deb https://build.opensuse.org/repositories/devel:kubic:libcontainers:stable:cri-o:1.17:1.17.3 /' > /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list" &&     curl -LO https://download.opensuse.org/repositories/devel:kubic:libcontainers:stable/xUbuntu_19.10/Release.key &&     apt-key add - < Release.key && apt-get update &&     apt-get install -y --no-install-recommends cri-o-1.17' returned a non-zero code: 100
make: *** [kic-base-image] Error 100
```

**Comment by haircommander** [comment]
you need to also add the top level repository:
```bash
echo "deb https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/$OS/ /" > /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list
curl -L https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/$OS/Release.key | apt-key add -
apt install containers-common
```

**Comment by tstromberg** [comment]
Perhaps I am doing this wrong, but the v1.17.3~3 debian package seems to be missing nearly all of the files it should contain, such as a binary or systemd file:

# sudo dpkg -L cri-o
/.
/usr
/usr/share
/usr/share/doc
/usr/share/doc/cri-o
/usr/share/doc/cri-o/changelog.gz
/usr/share/doc/cri-o/copyright
```

I note that the package within 
https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable:/cri-o:/1.17:/1.17.3/xUbuntu_19.10/amd64/ is only 5.5KB.

The package within https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable:/cri-o:/1.17:/1.17.4/xUbuntu_20.04/amd64/ is 20MB.

**Comment by tstromberg** [comment]
The other quirk worth noting, in case it helps isolate the root cause, is that the v1.17.4 package is called 'cri-o-1.17', and the v1.17.3 package is called 'cri-o'.

**Comment by haircommander** [comment]
I would sooner believe that I packaged it incorrectly than that 15mb difference is expected. I will try to look into it later today or tomorrow, kind of swamped rn

**Comment by tstromberg** [comment]
Thank you for the update!

The inability to build a Docker image using crio v1.17.3 is blocking our minikube release, so we'll try in parallel to upgrade to v1.18.x to see if it addresses the strange disappearing pods after restart behavior we're seeing in v1.17.4

**Comment by haircommander** [comment]
as a side note, I'd like to hear more about this strange behaviour in 1.17.4, but I believe that's worthy of a new issue

**Comment by tstromberg** [comment]
> as a side note, I'd like to hear more about this strange behaviour in 1.17.4, but I believe that's worthy of a new issue

Done: https://github.com/cri-o/cri-o/issues/4035

**Comment by afbjorklund** [comment]
@haircommander

> I am not sure where 1.17.3 went

I think this is a limitation of OBS, we have the same problem with podman...

The only way to use this repository reliably is to cache the packages "locally".

**Comment by haircommander** [comment]
you can pin to a release by setting `$VERSION` to `1.17:1.17.3`
that allows you to pin to a specific release, which will not update with the 1.17 project
https://build.opensuse.org/project/show/devel:kubic:libcontainers:stable:cri-o:1.17:1.17.3

**Comment by haircommander** [comment]
the subproject 1.17.3 is created, can y'all try it out and let me know if it solves your problems?

**Comment by afbjorklund** [comment]
It seems that crio was already upgraded to 1.18.3 instead: https://github.com/kubernetes/minikube/commit/61583d62492e2be0387563f460c56b1afba264c1

**Comment by afbjorklund** [comment]
Prepared both 1.17.3 and 1.18.3 (requires new OS/go), as an upgrade to current 1.17.1
